### ISSUE TITLE
Allow @onready to be used with static variables or add @deferred tag that allows static variables to be initialized later

### ISSUE THREAD

**Issue by filipworksdev** [godotengine/godot-proposals#10376]
### Describe the project you are working on

A game with a lot of static variables

### Describe the problem or limitation you are having in your project

`static var myNode = get_node("MyNode")`

This code will not work because you cannot call non static methods in a static initializer

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Suggestion 1:

Allow using `@onready` for static variables to delay the initialization when node is ready

`@onready static var myNode = $MyNode`

This is same as setting the static var later in `_ready()`

Suggestion 2:

The other idea is to create a `@deferred` tag which allows anything to be initialized later including static variables.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

This will make the code easier to write and understand.

Currently I can do this

```
static var node

func _ready():
    node = get_node("Node")
```

This will become easier to write and understand

`@onready var node = get_node("Node")`

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can be worked around but a lot more lines of script yes.

### Is there a reason why this should be core and not an add-on in the asset library?

Needs to be a core change.

**Comment by OffsetMOSFET** [comment]
I don't think this has clear behavior if there are multiple instances of the script.

**Comment by filipworksdev** [comment]
You are right. I am guessing the behavior would be to initialize a static variable only once but yes it is a bit confusing.
